File: backend/src/main/java/com/bakery/domain/model/UnitOfMeasurement.java
package com.bakery.domain.model;

/**
 * Unidades de medida soportadas para insumos y rendimientos de receta.
 * Corresponden 1:1 con la tabla units_of_measurement (ids 1-6).
 */
public enum UnitOfMeasurement {

    KILOGRAM("kg"),
    GRAM("g"),
    MILLIGRAM("mg"),
    LITER("l"),
    MILLILITER("ml"),
    UNIT("u");

    private final String symbol;

    UnitOfMeasurement(String symbol) {
        this.symbol = symbol;
    }

    public String getSymbol() {
        return symbol;
    }
}

File: frontend/src/app/core/models/unit-of-measurement.ts
export enum UnitOfMeasurement {
  KILOGRAM = 'KILOGRAM',
  GRAM = 'GRAM',
  MILLIGRAM = 'MILLIGRAM',
  LITER = 'LITER',
  MILLILITER = 'MILLILITER',
  UNIT = 'UNIT',
}

export const UNIT_SYMBOLS: Record<UnitOfMeasurement, string> = {
  [UnitOfMeasurement.KILOGRAM]: 'kg',
  [UnitOfMeasurement.GRAM]: 'g',
  [UnitOfMeasurement.MILLIGRAM]: 'mg',
  [UnitOfMeasurement.LITER]: 'l',
  [UnitOfMeasurement.MILLILITER]: 'ml',
  [UnitOfMeasurement.UNIT]: 'u',
};

export const UNIT_OPTIONS = Object.values(UnitOfMeasurement).map((unit) => ({
  value: unit,
  label: `${unit} (${UNIT_SYMBOLS[unit]})`,
}));
